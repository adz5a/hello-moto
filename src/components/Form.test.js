import React from "react";
import {
    Button,
    InputWithLabel,
    Input
} from "./Form";
import { shallow } from Â "enzyme";


test("Form component", () => {

    [
        Button,
        InputWithLabel,
        Input
    ]
        .forEach( Component => {

            const tree = shallow(<Component />);
            expect(tree).toMatchSnapshot();

        } );

});
