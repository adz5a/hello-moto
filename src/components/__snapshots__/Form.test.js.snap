// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <input
    className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6"
    type="button"
    value={undefined}
/>,
  "nodes": Array [
    <input
      className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6"
      type="button"
      value={undefined}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Button />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <input
          className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6"
          type="button"
          value={undefined}
/>,
        "_debugID": 2,
        "_renderedOutput": <input
          className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6"
          type="button"
          value={undefined}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Button />,
}
`;

exports[`Form component 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <span>
    <label
        className="db fw4 lh-copy f6"
        htmlFor="--"
    />
    <input
        className="b pa2 input-reset ba bg-transparent"
        id="--"
        name={undefined}
        type={undefined}
    />
</span>,
  "nodes": Array [
    <span>
      <label
            className="db fw4 lh-copy f6"
            htmlFor="--"
      />
      <input
            className="b pa2 input-reset ba bg-transparent"
            id="--"
            name={undefined}
            type={undefined}
      />
</span>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <InputWithLabel />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <span>
          <label
                    className="db fw4 lh-copy f6"
                    htmlFor="--"
          />
          <input
                    className="b pa2 input-reset ba bg-transparent"
                    id="--"
                    name={undefined}
                    type={undefined}
          />
</span>,
        "_debugID": 4,
        "_renderedOutput": <span>
          <label
                    className="db fw4 lh-copy f6"
                    htmlFor="--"
          />
          <input
                    className="b pa2 input-reset ba bg-transparent"
                    id="--"
                    name={undefined}
                    type={undefined}
          />
</span>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InputWithLabel />,
}
`;

exports[`Form component 3`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <input
    className="b pa2 input-reset ba bg-transparent"
    name={undefined}
    type={undefined}
/>,
  "nodes": Array [
    <input
      className="b pa2 input-reset ba bg-transparent"
      name={undefined}
      type={undefined}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Input />,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <input
          className="b pa2 input-reset ba bg-transparent"
          name={undefined}
          type={undefined}
/>,
        "_debugID": 6,
        "_renderedOutput": <input
          className="b pa2 input-reset ba bg-transparent"
          name={undefined}
          type={undefined}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Input />,
}
`;
